game/src/graphic/textures/TextureHandler.java

```diff
@@ -2,31 +2,70 @@
 
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.files.FileHandle;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Arrays;
+import java.util.Comparator;
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.function.Predicate;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 /**
  * A texture handler class for managing textures paths for further use.
  *
- * <p>Singleton (possibly thread-safe).
+ * <p>This class is designed as Singleton, because all asset paths should be read only once at the
+ * beginning of the application.
  */
 public class TextureHandler {
-    private static final TextureHandler INSTANCE = new TextureHandler();
+    protected static final String PLACEHOLDER_FILENAME = ".resource_root";
+
+    /** See also: {@link TextureHandler#getResourceRoot()} */
+    private static final int maxDepth = 4;
+
+    private static TextureHandler INSTANCE;
 
     private final Map<String, Set<FileHandle>> pathMap = new LinkedHashMap<>();
 
-    private TextureHandler() {
-        addAllAssets(Gdx.files.internal(Gdx.files.getLocalStoragePath()));
+    private TextureHandler() throws IOException {
+        addAllAssets(getResourceRoot());
+    }
+
+    /**
+     * Searches for the resource root that has the longest path name string.
+     *
+     * <p>{@link TextureHandler#maxDepth}: We assume that the build directory structure looks like
+     * this: "Dungeon/<build dir>/resources/main/<...>". So we need to dive at least into the
+     * directories with a depth of 4.
+     *
+     * @return the resource root with the longest path name string.
+     */
```

Comment:
Dieser Javadoc ist nicht sichtbar (per Default wg. `private`). Kannst Du den Kommentar bitte in den Klassen-Javadoc übertragen?